scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java

```diff
@@ -300,8 +297,48 @@ public String getVirtualHost() {
 		return virtualHost;
 	}
 
-	void setVirtualHost(String hostname) {
+	/**
+	 * Set the (virtual) host name for the server that this session has been
+	 * established for.
+	 * <p>
+	 * 
+	 * @param hostname the virtual host name at the peer (may be {@code null}).
+	 */
+	public void setVirtualHost(String hostname) {
+		this.serverNames = null;
 		this.virtualHost = hostname;
+		if (hostname != null) {
+			this.serverNames = ServerNames
+					.newInstance(ServerName.from(NameType.HOST_NAME, hostname.getBytes(ServerName.CHARSET)));
+		}
+	}
+
+	/**
+	 * Gets the server names for the server that this session
+	 * has been established for.
+	 * 
+	 * @return server names, or {@code null}, if not used.
+	 */
+	public ServerNames getServerNames() {
+		return serverNames;
+	}
+
+	/**
+	 * Set the server names for the server that this session has been
+	 * established for.
+	 * <p>
+	 * 
+	 * @param serverNames the server names (may be {@code null}).
+	 */
+	public void setServerNames(ServerNames serverNames) {
+		this.virtualHost = null;
+		this.serverNames = serverNames;
```

Comment:
I'm not so aware about SNI extension but there is several server names for a given session ?

Reply:
Several of different types. But the hostname is only one type and currently the only supported :-).